### PR TITLE
Deprecate is_pyqt5.

### PR DESCRIPTION
**Pull request by anntzer**
The function won't really make sense anymore after the release of
PyQt6.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_qt5.py [hunk 1]
```diff
@@ -129,11 +129,10 @@ def _create_qApp():
         else:
             qApp = app
 
-    if is_pyqt5():
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_qt5.py]: the check is redundant with the try... except AttributeError.

**Comment by QuLogic** [comment]
Much like the Python 3 transition, do we want to check for >=5 instead? Otherwise, PyQt6 will use the PyQt4 code, but I assume the PyQt6 release will be somewhat closer to the 5 version than 4.

**Comment by anntzer** [comment]
sounds good.

File: doc/api/api_changes_3.3/deprecations.rst [hunk 2]
```diff
@@ -535,3 +535,8 @@ These are unused and can be easily reproduced by other date tools.
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 The ``cbid`` and ``locator`` attribute are deprecated.  Use
 ``mappable.colorbar_cid`` and ``colorbar.locator``, as for standard colorbars.
+
+``qt_compat.is_pyqt5``
+~~~~~~~~~~~~~~~~~~~~~~
+This function is deprecated in prevision of the future release of PyQt6.  The
+Qt version can be checked using ``QtCore.qVersion()``.
```
Comment (QuLogic) [inline doc/api/api_changes_3.3/deprecations.rst:542]: Or `QtCore.QT_VERSION_STR`?

Comment (anntzer) [reply doc/api/api_changes_3.3/deprecations.rst:542]: yes

**Pull request merged**